lang/python/python3/patches-pip/001-pep517-pyc-fix.patch

```diff
@@ -0,0 +1,12 @@
+diff -Nurp a/pip/_vendor/pep517/wrappers.py b/pip/_vendor/pep517/wrappers.py
+--- a/pip/_vendor/pep517/wrappers.py	2019-07-30 20:02:13.000000000 +0800
++++ b/pip/_vendor/pep517/wrappers.py	2020-04-23 23:01:10.022472889 +0800
+@@ -8,7 +8,7 @@ from tempfile import mkdtemp
+ 
+ from . import compat
+ 
+-_in_proc_script = pjoin(dirname(abspath(__file__)), '_in_process.py')
++_in_proc_script = pjoin(dirname(abspath(__file__)), '_in_process.pyc')
```

Comment:
should we maybe fine a solution that works best for both target and host?
keeping separate patches for host & target sounds like it could become a maintenance  pain;
for this patch, maybe we could do something like:
```
_in_proc_script = None
for ext in [ '.py', '.pyc' ]:
    script = pjoin(dirname(abspath(__file__)), '_in_process.pyc')
    if (os.path.exist(script)):
        _in_proc_script = script;

if (_in_proc_script is None):
    throw some exception
```

maybe we could also push this upstream:
https://github.com/pypa/pip/blob/master/src/pip/_vendor/pep517/wrappers.py#L16

it looks like pip will change some context there, so this patch will need to be re-applied on our side, which will become annoying quick

Reply:
I think maintaining separate patches for host and target would be easier than writing all patches to work for both :joy:

In this instance, it would mean changing a 1 line patch into 7 lines, and it has no practical benefit for host pip.

Also I don't think upstream would be interested in a patch like this, because I believe we are the only distro that distributes .pyc files without the .py source file, and because we are using the legacy format (.pyc file in the same directory as .py file) whereas on other platforms the actual .pyc file is `__pycache__/_in_process.cpython-38.pyc`.

Reply:
>  and because we are using the legacy format

i keep fearing the day they drop the legacy format;

no idea how this works these days, but there are a few more other formats ; there's`.pyc` `.pyd`, `.pye`
i forget what `.pyd` is;
`.pye` is encrypted; no idea how that is being distributed; and is rare;


in any case, even a 3-4 line patch is a bit better than patches for host & patches for target;
so; for this, we could just do:
```
import os
_in_proc_script = pjoin(dirname(abspath(__file__)), '_in_process.pyc')
if (not os.path.exists(_in_proc_script))):
    _in_proc_script = pjoin(dirname(abspath(__file__)), '_in_process.py')
```

Reply:
If we flip the order of .pyc/.py then this has the advantage that if the user installed the source package and made modifications to the .py file, then their modifications will be used.

Ok you win :joy: since this is only one patch and it is very simple, I'll rework the PR to only add this patch and not split the patch directories.